src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,21 @@
+import Popup from "./Popup";
+
+export default class PopupWithConfirm extends Popup {
+  constructor(popupSelector) {
+    super({ popupSelector });
+    this._submitButton = this._popupElement.querySelector(".modal__save");
+    this._handleSubmitCallback = null;
+    this._cardId = null;
```

Comment:
`    this._cardId = null;
`

You don't reassign this field, so you don't need it